src/ast/isReference.js

```diff
@@ -4,6 +4,8 @@ export default function isReference ( node, parent ) {
 	}
 
 	if ( node.type === 'Identifier' ) {
+		if ( !parent ) return true;
```

Comment:
It's true that only `Program` doesn't have a parent, but as far as the enter handler goes it depends where you start the walk – in this situation we're starting at the function body. Ordinarily you couldn't reach an identifier node without first going through its parent, but there's an exception in this case, because an identifier can form the entire body of an arrow function expression. Adding a comment to the code